Test/civl/inductive-sequentialization/paxos/Paxos.bpl

```diff
@@ -48,31 +48,13 @@ function {:pending_async "A_Propose"}{:constructor} Propose_PA(round: Round, ps:
 function {:pending_async "A_Vote"}{:constructor} Vote_PA(round: Round, node: Node, value: Value, p: Permission) : PA;
 function {:pending_async "A_Conclude"}{:constructor} Conclude_PA(round: Round, value: Value, p: Permission) : PA;
 
-////////////////////////////////////////////////////////////////////////////////
-//// Generalized array theory imports
-
-function {:builtin "MapConst"} MapConstPermission(bool): [Permission]bool;
-function {:builtin "MapOr"} MapOrPermission([Permission]bool, [Permission]bool): [Permission]bool;
-function {:builtin "MapConst"} MapConstNode(bool): NodeSet;
-function {:builtin "MapOr"} MapOrNode(NodeSet, NodeSet): NodeSet;
-function {:builtin "MapAnd"} MapAndNode(NodeSet, NodeSet): NodeSet;
-function {:builtin "MapImp"} MapImpNode(NodeSet, NodeSet) : NodeSet;
-function {:builtin "MapConst"} MapConstPA(int): [PA]int;
-function {:builtin "MapAdd"} MapAddPA([PA]int, [PA]int): [PA]int;
-function {:builtin "MapSub"} MapSubPA([PA]int, [PA]int): [PA]int;
-function {:builtin "MapConst"} MapConstBool(bool) : [Permission]bool;
-function {:builtin "MapOr"} MapOr([Permission]bool, [Permission]bool) : [Permission]bool;
-function {:builtin "MapConst"} MapConstVoteResponse(int): [VoteResponse]int;
-function {:builtin "MapAdd"} MapAddVoteResponse([VoteResponse]int, [VoteResponse]int): [VoteResponse]int;
-function {:builtin "MapSub"} MapSubVoteResponse([VoteResponse]int, [VoteResponse]int): [VoteResponse]int;
-
 ////////////////////////////////////////////////////////////////////////////////
 //// Functions
 
-function {:inline} NoPAs(): [PA]int { MapConstPA(0) }
+function {:inline} NoPAs(): [PA]int { MapConst(0) }
 function {:inline} SingletonPA(pa:PA): [PA]int { NoPAs()[pa := 1] }
```

Comment:
In addition to singleton sets (currently called `MapUnit`), it would be nice to introducing polymorphic functions for empty sets as well as empty and singleton multisets. Then it would be possible to eliminate quite some more boilerplate definitions. But this can also be done in another PR.

Reply:
We already have the polymorphic function for "empty set" (MapConst(false)) and "empty multiset" MapConst(0). I hesitate to call these things sets and multisets because they are allowed to be infinite.  I was hoping to reserve the words set and multiset for the finite versions that will be introduced later.  

For singleton, would it be fine to introduce the following generic function:

function {:inline} MapAllExcept<T,U>(t: T, u: U, default: U): [T]U {
  MapConst(default)[t := u]
}